firefox-ios/Client/Frontend/TabContentsScripts/LoginsHelper.swift

```diff
@@ -273,18 +344,28 @@ class LoginsHelper: TabContentScript {
                 return login?.httpRealm == nil ? login?.toJSONDict() : nil
             }
 
-            let dict: [String: Any] = [
-                "requestId": requestId,
-                "name": "RemoteLogins:loginsFound",
-                "logins": logins
-            ]
-            guard let injected = dict.asString else { return }
-            let injectJavaScript = "window.__firefox__.logins.inject(\(injected))"
-            self.tab?.webView?.evaluateJavascriptInDefaultContentWorld(injectJavaScript)
-            self.sendLoginsAutofilledTelemetry()
+            let loginData = LoginInjectionData(requestId: self.currentRequestId,
+                                               logins: logins)
+
+            // NOTE: FXIOS-3856 This will get disabled in future with addtion of bottom sheet
+            guard let tab = self.tab else {
+                return
+            }
+
+            LoginsHelper.fillLoginDetails(with: tab, loginData: loginData)
         }
     }
 
+    public static func fillLoginDetails(with tab: Tab,
+                                        loginData: LoginInjectionData) {
+        guard let injected = loginData.toJSONString() else { return }
```

Comment:
@razvanlitianu 
PART2: This will be used to inject password into the tab or webpage you are on

Basically in your bottom sheet as long as you have all the data related to passwords you should be able to use this method to inject it by creating the required object. This also means we will do some cleanup as mentioned in the requestLogins method above and remove that piece. 

Note: There is a chance we might need `currentRequestId:` but will take care of that when needed